### DISCUSSION TITLE
v4.0.35 pre-release

### DISCUSSION THREAD

**Discussion by autoSteve** [BJReplay/ha-solcast-solar#69] [General]
I've released v4.0.35 as a pre-release.

The silence of discussion for v4.0.34 was profound (#63). High five.

To give this pre-release a go, please make sure that `Show beta versions` is on when re-downloading from HACS.

I've also added a 'nudge' in the readme to try the beta version if folks are willing. That might expand the number of testers. Huge thanks to the known testers of v4.0.34, @gcoan, @ProphetOfDoom, @Nilogax, @iainfogg, and probably @robinhood-code.

As always, both positive and negative feedback is most welcome, and this will contribute to getting the release ultimately published as latest. Report back, and don't be shy!

> [!NOTE]
> The BJReplay fork is still not published for searching in HACS, so continued use of a manual install in HACS is the go. As explained earlier we're going to keep any releases as pre-release on the fork until that happens, as any 'latest' version change would require modification of that HACS pull request.

## What's Changed
* (Enhancement) Breakdown of individual site forecast wattage and time as attributes by @autoSteve
* Do not log options version upgrade if no upgrade is required by @autoSteve
* Add info about preserving oziee history and config to banner by @iainfogg

## New Contributors
* @iainfogg made their first contribution 🎉 in https://github.com/BJReplay/ha-solcast-solar/pull/67

**Full Changelog**: https://github.com/BJReplay/ha-solcast-solar/compare/v4.0.34...v4.0.35

**Comment by autoSteve** [comment]
Eleven downloads. Good start! Don't pull with the punches, and don't be afraid to sing "meh, it works flawlessly..."

**Reply by gcoan** [reply]
Installed 4.0.35, nothing untoward in the debug log (I see you crept in a new Info log of the Solcast version).

All sensors and the site level peak data appear to be working fine.

I'll add a further update when my next scheduled poll runs, but for now, its all lovely ❤️ and roses 🌹

**Reply by gcoan** [reply]
Scheduled poll at 18:00 all looks good.  Sensors updated, nothing untoward in the logfile, files written as expected.  All good and nothing to see here

**Reply by EddieSpaghettie** [reply]
Works Well. No Issues!

**Comment by dtswk** [comment]
Thanks @autoSteve working perfectly for me, I love the additional rolled up estimates per day (P10 and P90), you just retired a bunch of my template code  :-)   I'm so impressed I am going to end my node-red direct downloads and calcs and just pass all this back via MQTT.

**Reply by autoSteve** [reply]
> rolled up estimates per day (P10 and P90)

Great to hear. 🥰

`*` Coming soon to a v4.0.36 near you. (Spoiler alert!) @dtswk is running some alpha v4.0.next code based on v4.0.35 that exposes additional sensor attributes. #51.

**Comment by ProphetOfDoom** [comment]
Ahhh. The sweet smell of a version update in the morning!

solcast.json and solcast-usage.json files updated. All sensors populated including nice new ones such as "Power Next 30 Mins" and roof tally values match values in solcast.json file. No solcast related errors in log.

Looking forward to Release V5.0.0 sometime to make an obvious clean break with the past.

Hugs to all

**Comment by Nilogax** [comment]
Meh, it works flawlessly... (so far!)

Thanks again to all for the ongoing development

**Comment by dtswk** [comment]
Ok, all updated this morning and ... yawn.... nothing to see here, works perfectly (goes off to write his own interesting aka broken code )

**Comment by autoSteve** [comment]
Thirty-five downloads. Who would have thought? People actually _do_ read the readme... 😘

**Reply by gcoan** [reply]
35 new friends Steve. You've got Oziee to thank for that

**Reply by autoSteve** [reply]
43 new ones now. 🤦🏻

But how's the silence? (knocks on wood...)

**Comment by wimdebruyn** [comment]
No problems and working well. Thank you!

**Reply by autoSteve** [reply]
Naww 🥰. A new beta friend. Glad to hear from you @wimdebruyn.

**Comment by autoSteve** [comment]
There are now fifty three wonderful humans running this beta. To all, a good night from me, and happy solar forecasting.

**Reply by autoSteve** [reply]
Eighty-four downloads for v4.0.35.

Enjoying the silence...
